lib/log_file.c

```diff
@@ -132,13 +148,14 @@ int32_t
 qb_log_file_reopen(int32_t t, const char *filename)
 {
 	struct qb_log_target *target = qb_log_target_get(t);
+	int rc;
 
 	if (filename) {
 		(void)strlcpy(target->filename, filename, PATH_MAX);
 	}
-	_file_reload(t);
+	rc = _do_file_reload(t);
 
-	if (!target->instance) {
+	if (rc) {
 		return -errno;
 	} else {
 		return 0;
```

Comment:
Why not move this part into `_do_file_reload()` so that you can simply `return _do_file_reload(t)` here? Gets rid of a conditional and moves reading of `errno` much closer to the function setting it, which adds to clarity IMO.

Reply:
We can't do that because _do_file_reload() is called from the logger object and that doesn't return an error (bad design, I know, not guilty). That's why I had to split it into two functions in the first place!

Reply:
You mean `_file_reload()`, don't you? You could still define it to `errno = -_do_file_reload(t, NULL)` I guess (using the second argument for passing in the new filename as discussed in my other comment). Then get rid of it altogether with the next API change.